### ISSUE TITLE
New rule suggestion: flag identical stacked decorators

### ISSUE THREAD

**Issue by jonathandung** [astral-sh/ruff#26042]
### Summary

Usually, stacking multiple `@overload`s on one overload may mean that there were overloads that were accidentally deleted, but the linter does not catch this. If that is not the case, the stacking would be unnecessary.

```python
from typing import overload
@overload
def process(num: int, /) -> str: ...
@overload # diagnostic can point to this line and note: was an overload deleted?
# Safe autofix: delete the extra instances of @overload
@overload
def process(ba: bytearray, /) -> memoryview[int]: ...
```

**Comment by jonathandung** [comment]
Alternatively, this can be a `ty` diagnostic.

**Comment by MichaReiser** [comment]
This seems reasonable to me. I also think it's fine to have in Ruff. The only question here is whether this should be specific to `overload` or more a: known decorators that should only be applied once (e.g `@override` is another one)

**Comment by jonathandung** [comment]
`staticmethod`, `classmethod`, `property` as well

**Comment by jonathandung** [comment]
In fact, I can't think of a case where stacking identical decorators would make sense. Maybe this could be a catch-all lint with configurable known exclusions.

**Comment by MichaReiser** [comment]
There are multiple decorators that allow stacking

```py
# Cartesian product of parameters
@pytest.mark.parametrize("x", [0, 1])
@pytest.mark.parametrize("y", [2, 3])
def test_something(x, y): ...
```

[pytest](https://docs.pytest.org/en/stable/how-to/parametrize.html).

```py
# Patch two different objects
@patch("package.client")
@patch("package.clock")
def test_something(mock_clock, mock_client): ...
```

The standard library [documents nested patch decorators](https://docs.python.org/3/library/unittest.mock.html#nesting-patch-decorators).


```py
# Register one handler for two routes
@app.route("/")
@app.route("/index")
def index(): ...
```

https://flask.palletsprojects.com/en/stable/patterns/singlepageapplications/

```py
handlers = []

def register(function):
    handlers.append(function)
    return function

@register
@register
def handle_event(): ...
```

That's why I think an allowlist is probably better here

**Comment by jonathandung** [comment]
I meant identical as in treating passing different parameters to decorator factories as different, but I didn't think of the last case.
